packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -4849,6 +4853,9 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   @optionalTypeArgs
   Future<T?> push<T extends Object?>(Route<T> route) {
     _pushEntry(_RouteEntry(route, pageBased: false, initialState: _RouteLifecycle.push));
+    // if (route is FlexibleTransitionRouteMixin<T>) {
+    //   delegateTransitionBuilder = (route as FlexibleTransitionRouteMixin<T>).delegatedTransition;
+    // }
```

Comment:
@justinmc this code ended up being problematic for train hopping, due to the fact that it gets picked up by all pages on the screen. So if a MBS route is opened, then mid transition another page gets pushed, both the MBS route and it's parent route pick up this delegate transition. Our normal logic is able to handle the MBS route, but the original parent route is unable to adjust.

As is, this PR is able to work as expected and handle route changes mid transitions, but by using side effects.